rust/cubesql/cubesql/src/compile/rewrite/rules/filters.rs

```diff
@@ -2253,6 +2253,59 @@ impl RewriteRules for FilterRules {
                     "FilterOpOp:and",
                 ),
             ),
+            transforming_rewrite(
```

Comment:
We shouldn't add new rules here. If two rules like these are introduced previous two should be adjusted to use minus one millisecond logic so all existing tests are still passing.

Reply:
Ahh, I see. To make sure I'm following, my takeaway is that the regressions are coming from the `date_range` filter not matching.  This change only matches when both sides of the SQL is <= and >=. So there would need to be additional filters for <= and > and vice versa. 

One pointer I would really appreciate is how to incorporate the minus_one logic with the filter?

I.e. I want to add the following filter:
```
transforming_rewrite(
                "filter-replacer-in-date-range-inclusive-two",
                filter_op(
                    filter_op_filters(
                        filter_member(
                            "?member",
                            "FilterMemberOp:afterOrOnDate",
                            "?date_range_start",
                        ),
                        filter_member(
                            "?member",
                            "FilterMemberOp:beforeDate",
                            "?date_range_end", <-- Here, I want to call a minus_one transformation
                        ),
                    ),
                    "FilterOpOp:and",
                ),
                filter_member("?member", "FilterMemberOp:inDateRange", "?date_range"),
                self.merge_date_range("?date_range_start", "?date_range_end", "?date_range"),
            ),
```

I'm struggling to figure out what the minus_one transformation should look like here

Reply:
The dumb solution I can think of is to add in a transformation param into `self.merge_date_range("?date_range_start", "?date_range_end", "?date_range")` and do the transformation in merge_date_range.

i.e.
`self.merge_date_range("?date_range_start", "?date_range_end", "?date_range", 'TransformMemberOp:StartMinusOne)`
`self.merge_date_range("?date_range_start", "?date_range_end", "?date_range", 'TransformMemberOp:StartPlusOne)`
`self.merge_date_range("?date_range_start", "?date_range_end", "?date_range", 'TransformMemberOp:EndMinusOne)`
`self.merge_date_range("?date_range_start", "?date_range_end", "?date_range", 'TransformMemberOp:EndPlusOne)`

Reply:
@darapuk I'd actually take a slightly bit different approach. In fact `filter-replacer-in-date-range-inclusive` rule should become smart enough to match either `afterDate` or `afterOrOnDate` and `beforeDate` or `beforeOrOnDate` and coerce it to `inDateRange` with minus/plus ms logic. To avoid cartesian products of all possible inversions it should also understand if `afterDate` comes first or last. So it'll be the one rule to rule them all.

Reply:
I logically understand where you're going, but need that hint for how to extract the "operation" from the SQL. In `filter-replacer-in-date-range-inclusive` the filter_member seems to match exactly on the filter type:
```
let op = match_data_node!(node_by_id, params[1], FilterMemberOp);
```

What I'm not following is how to extract which type of operator is being used then pass it into the transformer.

I.e. 
What I want to be able to write is something like:
```
            transforming_rewrite(
                "filter-replacer-in-date-range-inclusive",
                filter_op(
                    filter_op_filters(
                        filter_member(
                            "?member",
                            "?left_operator",
                            "?date_range_left",
                        ),
                        filter_member(
                            "?member",
                            "?right_operator",
                            "?date_range_right",
                        ),
                    ),
                    "FilterOpOp:and",
                ),
                filter_member("?member", "FilterMemberOp:inDateRange", "?date_range"),
                self.merge_date_range("?date_range_left", "?date_range_right", "?date_range", "?right_operator", "left_operator"),
```

is that at all close, or am I totally misunderstanding how these filters are working?